src/azure-cli-core/azure/cli/core/telemetry.py

```diff
@@ -663,6 +672,17 @@ def _get_shell_type():
     return _remove_cmd_chars(_remove_symbols(os.environ.get('SHELL')))
 
 
+@decorators.suppress_all_exceptions(fallback_return='')
+def _get_user_agent():
+    if _session.user_agent:
+        return _session.user_agent
+    from azure.cli.core.util import get_az_user_agent
+    agents = [get_az_user_agent()]
+    if 'AZURE_HTTP_USER_AGENT' in os.environ:
+        agents.append(os.environ['AZURE_HTTP_USER_AGENT'])
+    return ' '.join(agents)
```

Comment:
When there's no user agent logged (Usually it's because no http request for some specific cli commands), we still record user agent info containing cli version info and env variable value here as requested by @MoChilia